### DISCUSSION TITLE
Available IOPS

### DISCUSSION THREAD

**Discussion by wyahn1219** [NetApp/harvest#2103] [Q&A] [answered]
Hello 
Can Harvest and Garafana check the Available IOPS counter like in Unified Manager?

**Comment by rahulguptajss** [comment]
@wyahn1219 Harvest captures IOPS and many more metrics like latency, amount of data transferred etc. See the volume dashboard screenshot in Harvest.

<img width="1387" alt="image" src="https://github.com/NetApp/harvest/assets/25551691/6ef5eebc-b7de-48e4-abca-08530f8e55d8">

**Comment by wyahn1219** [comment]
@rahulguptajss 

The dashboard does not show information about available IOPS.

https://docs.netapp.com/us-en/active-iq-unified-manager-97/performance-checker/concept-what-available-iops-is.html
Is there any way to view this information?

**Comment by rahulguptajss** [comment]
Thanks @wyahn1219 . I believe these are headroom counters. Harvest provides headroom dashboard that displays this information.

<img width="1386" alt="image" src="https://github.com/NetApp/harvest/assets/25551691/ee502b92-58dd-45ea-b765-449fcc23919a">

**Comment by wyahn1219** [comment]
Thank you for your answer.
I see very low available IOPS even though I rarely use aggr.
is this right?

headroom_aggr_optimal_point_ops{datacenter=~"$Datacenter",cluster=~"$Cluster",aggr=~"$OptimalPointAggr"} - headroom_aggr_current_ops{datacenter=~"$Datacenter",cluster=~"$Cluster",aggr=~"$OptimalPointAggr"}

**Comment by rahulguptajss** [comment, marked as answer]
Are you seeing it where confidence_factor is over 1? There was a similar problem reported in issue #1625.

headroom_aggr_optimal_point_confidence_factor{datacenter=~"$Datacenter",cluster=~"$Cluster"} > 1

<img width="1244" alt="image" src="https://github.com/NetApp/harvest/assets/25551691/f20cfdc3-45b1-47b3-9f63-3f431dda7387">

**Comment by wyahn1219** [comment]
No I don't see 
![image](https://github.com/NetApp/harvest/assets/103977808/2937cb80-e3ed-4d32-87ce-59b8a03c9de4)

**Comment by rahulguptajss** [comment]
Could you check #1625? It contains specific information indicating that the calculation for Available Ops is only performed when the value of _headroom_aggr_optimal_point_confidence_factor_ exceeds 1. Are you suggesting that even when the confidence factor is above 1, the reported available ops values are incorrect in dashboard?

**Comment by wyahn1219** [comment]
um.. what is ```headroom_aggr_optimal_point_confidence_factor``` ?
What do I need to do to make this return 1?

**Comment by rahulguptajss** [comment]
For more detailed information regarding this topic, you can refer to the following resource: [Is my controller overloaded?](https://kb.netapp.com/onprem/ontap/Performance/Is_my_controller_overloaded%3F). According to my understanding, this is calculated on ONTAP side. However, I don't have specific knowledge about how exactly this calculation is carried out in ONTAP.

<img width="1299" alt="image" src="https://github.com/NetApp/harvest/assets/25551691/f0b04fde-e331-4020-a781-01f7e475dd5a">

**Comment by rahulguptajss** [comment]
Please refer here for Zapi/Rest mapping for this counter https://netapp.github.io/harvest/23.05/ontap-metrics/#headroom_aggr_optimal_point_confidence_factor

**Comment by wyahn1219** [comment]
Oops sorry your already provide #1625.
Thank you for your kindly help.
